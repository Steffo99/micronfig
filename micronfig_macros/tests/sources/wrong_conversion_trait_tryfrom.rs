micronfig::config! {
	GARASAUTO: String => u64,
}

fn main() {
	println!("{:?}", GARASAUTO())
}
