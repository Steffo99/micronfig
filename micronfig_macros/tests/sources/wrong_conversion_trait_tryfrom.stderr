error[E0277]: the trait bound `u64: From<String>` is not satisfied
 --> tests/sources/wrong_conversion_trait_tryfrom.rs:1:1
  |
1 | / micronfig::config! {
2 | |     GARASAUTO: String => u64,
3 | | }
  | | ^
  | | |
  | |_the trait `From<String>` is not implemented for `u64`
  |   in this macro invocation
  |
 ::: src/lib.rs
  |
  |   pub fn config(input: TokenStream) -> TokenStream {
  |   ------------------------------------------------ in this expansion of `micronfig::config!`
  |
  = help: the following other types implement trait `From<T>`:
            <u64 as From<bool>>
            <u64 as From<char>>
            <u64 as From<u8>>
            <u64 as From<u16>>
            <u64 as From<u32>>
            <u64 as From<NonZeroU64>>
  = note: required for `String` to implement `Into<u64>`
  = note: required for `u64` to implement `TryFrom<String>`
  = note: required for `String` to implement `TryInto<u64>`
